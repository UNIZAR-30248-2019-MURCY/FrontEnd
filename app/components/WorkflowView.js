import React, {Component} from 'react';
import {ActivityIndicator, AsyncStorage, FlatList, StyleSheet, TextInput, View,} from 'react-native'
import {Button, ListItem, Text} from 'react-native-elements';

export default class WorkflowView extends Component {

    constructor(props) {
        super(props);
        this.state = {
            workflow: [],
            workflow2: [
                {
                    id:1,
                    description:'prueba',
                    response: 'pruebar'
                }
            ],
        }
    }


    componentDidMount() {
        if (this.props.navigation) {
            console.log(this.props.navigation.getParam('workflow', 'default value'));
            this.setState({workflow: this.props.navigation.getParam('workflow', 'default value')});
        }
    }

    render() {
        return (
            <View style={styles.container}>
                <View style={styles.containerTitle}>
                    <Text h2>Workflow</Text>
                </View>
                <View style={styles.containerWorkflow} className='reqList'>
                    <FlatList
                        className='flatList'
                        data={this.state.workflow}
                        keyExtractor={item => item.id.toString()}
                        renderItem={({item}) => (
                            <ListItem
                                containerStyle={{width: 300}}
                                className='detail-button'
                                title={ 'Description: ' + item.description}
                                titleStyle={{fontSize: 18}}
                                subtitle={item.response ? 'Response: ' + item.response + '\n' + 'Status: ' + item.status :  'Status: ' + item.status }
                                bottomDivider
                            />
                        )}
                    />
                </View>
                <View style={styles.containerReturn}>
                    <Button
                        className='return-button'
                        type="clear"
                        buttonStyle={styles.button2}
                        title="Return"
                        titleStyle={{color: 'grey', fontSize: 20}}
                        onPress={() => {
                            this.props.navigation.goBack();
                        }
                        }/>
                </View>
            </View>
        )
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        padding: 20
        //justifyContent: 'center',
    },
    horizontal: {
        flexDirection: 'row',
        justifyContent: 'space-around',
        padding: 10,
        marginTop: 50,
    },
    containerTitle: {
        alignItems: 'center',
        marginTop: 50,
        padding: 20
    },
    containerWorkflow: {
        flex: 1,
        marginTop: 20,
        alignItems: 'center',
    },
    containerReturn: {
        alignItems: 'center',
        marginBottom: 25
    },
})
