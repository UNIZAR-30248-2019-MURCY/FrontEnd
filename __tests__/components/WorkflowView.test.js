import React from 'react';
import {shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';
import WorkflowView from "../../../FrontEnd/app/components/WorkflowView";


describe('<WorkflowView />', () => {
    it('WorkflowView renders without crashing', () => {
        const rendered = renderer.create(<WorkflowView/>).toJSON();
        expect(rendered).toBeTruthy();
    });
    it('WorkflowView test against snapshot', () => {
        const page = renderer.create(<WorkflowView/>).toJSON();
        expect(page).toMatchSnapshot();
    });
});
