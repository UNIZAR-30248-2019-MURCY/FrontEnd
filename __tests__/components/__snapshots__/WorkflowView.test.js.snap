// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WorkflowView /> WorkflowView test against snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "padding": 20,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginTop": 50,
        "padding": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 42.5,
        }
      }
      theme={
        Object {
          "colors": Object {
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": Object {
              "android": Object {
                "error": "#f44336",
                "primary": "#2196f3",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "ios": Object {
                "error": "#ff3b30",
                "primary": "#007aff",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#8F0CE8",
            "success": "#52c41a",
            "warning": "#faad14",
          },
        }
      }
    >
      Workflow
    </Text>
  </View>
  <View
    className="reqList"
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "marginTop": 20,
      }
    }
  >
    <RCTScrollView
      className="flatList"
      data={Array []}
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View />
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginBottom": 25,
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 3,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "#2089dc",
              "borderRadius": 3,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "grey",
                "fontSize": 20,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            Return
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
